bot/cogs/grades_cog.py

```diff
@@ -0,0 +1,259 @@
+import json
+import os
+
+import logging
+import random
+import time
+
+import discord
+import discord.ext.commands as commands
+
+from bot.consts import Colors
+
+log = logging.getLogger(__name__)
+
+
+
+class gradesCog(commands.Cog):
+    special_converted_files = ['2018Fall.csv','2018Spring.csv'] #these files have special formatting for names (ie ' "Smith, John William" ' is normal and in these it's simply 'Smith John William' without commas or quotes )
+    #TBH, this is the easier way to parse but whatevs
+    converted_files = ['2013Fall.csv', '2014Spring.csv', '2014Fall.csv', '2015Spring.csv', '2015Fall.csv', '2016Spring.csv', '2016Fall.csv', '2017Spring.csv', '2017Fall.csv', '2019Spring.csv', '2019Fall.csv', '2020Spring.csv']
+
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.fileLines = []
+        self.master_list = {}
+        self.master_prof_list = {}
+
+
+
+    def process_Search(self, orig_query: str) -> str: #Primary search function and processing for a query. A query is generally defined by 
+                                    #"Course-Number" of which it pulls the data from the DB.
+        print("PROCESSING")
+        query = orig_query.upper()
+        
+        items = query.split("-")
+        master_String = ''
+
+        # if("ALL" in items):
+        #     print("ALL IN QUERY")
+        #     print("HERE COMES THE SUN")
+        #     if items[0] == "ALL" and items[1] == "ALL":
+        #         print("ALLALL")
+        #         for i in master_list:
+        #             master_String += process_Search(i)
+        #     elif items[0] == "ALL":
+        #         for i in master_list:
+        #             if items[1] in i:
+        #                 master_String += process_Search(i)
+        #     elif items[1] == "ALL":
+        #         for i in master_list:
+        #             if items[0] in i:
+        #                 master_String += process_Search(i)
+        #     else:
+        #         print("Misunderstood")
+        #         return "HELP"
+        #     return master_String
+
+        # else:
+        if query in self.master_list:
+            data_list = self.master_list[query]
+        else:
+            raise NotADirectoryError
```

Comment:
What is the purpose of this error? What does it represent in the bot

Reply:
if the class doesn't exist specifically

Reply:
Then that check should be brought out as a guard condition with an error embed at the start of the command